src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -271,6 +275,33 @@ message GetClusterStatusReply {
   ClusterResourceState cluster_resource_state = 2;
 }
 
+enum DrainNodeReason {
+  // The node has been idle for long enough time.
+  // Drain request with this reason is rejectable.
+  IDLE_TERMINATION = 0;
+  // The node will be preempted.
+  // Drain request with this reason is not rejectable.
+  PREEMPTION = 1;
+}
+
+message DrainNodeRequest {
```

Comment:
I make this request to drain only one node instead of having a batch version. Batch version will be more performant but it's more complicated for things like error handling.

Reply:
This should be a part of reporting autoscaler state.

Reply:
we need a field in reporting autoscaler state.